lom/src/lib/lomcommon/helper.go

```diff
@@ -151,6 +152,100 @@ func LogDebug(s string, a ...interface{})  {
     LogMessage(syslog.LOG_DEBUG, s, a...)
 }
 
+/*
+ * System Shutdown
+ *
+ */
+type sysShutdown_t struct {
+    ch          chan int
+    wg          *sync.WaitGroup
+    shutdown    bool
+}
+
+func (p *sysShutdown_t) register(caller string) <-chan int {
```

Comment:
How do we  ensure for each client string register and deregister must be called once respectively and in order.

Reply:
Answered above.